File: NativeWrappers/Windows/SimplySaveWindows/SimplySaveWindows/NotesStorage.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SimplySaveWindows
{
    public class NotesStorage
    {
        public string SaveFilesDir { get; private set; }

        public NotesStorage(string saveFilesDir)
        {
            SaveFilesDir = saveFilesDir;
        }

        public void CreateNote(string id, string content)
        {
            if (!Directory.Exists(SaveFilesDir))
                Directory.CreateDirectory(SaveFilesDir);

            string fullPath = Path.Combine(SaveFilesDir, $"{id}.ssnf");
            if (File.Exists(fullPath))
                throw new Exception($"Note with id {id} already exists! Probably, corrupted metadata.");

            using var sw = new StreamWriter(fullPath);
            sw.Write(content);
        }

        public void UpdateNote(string id, string content)
        {
            if (!Directory.Exists(SaveFilesDir))
                Directory.CreateDirectory(SaveFilesDir);

            string fullPath = Path.Combine(SaveFilesDir, $"{id}.ssnf");
            using var sw = new StreamWriter(fullPath);
            sw.Write(content);
        }

        public void DeleteNote(string id)
        {
            if (!Directory.Exists(SaveFilesDir))
                return;

            string fullPath = Path.Combine(SaveFilesDir, $"{id}.ssnf");
            if (File.Exists(fullPath))
                File.Delete(fullPath);
        }

        public string GetContent(string id)
        {
            string fullPath = Path.Combine(SaveFilesDir, $"{id}.ssnf");
            using var sr = new StreamReader(fullPath);
            return sr.ReadToEnd();
        }

    }
}

File: NativeWrappers/Android/SimplySaveAndroid/app/src/main/java/com/simplysave/simplysaveandroid/NotesStorage.kt
package com.simplysave.simplysaveandroid

import java.io.File

class NotesStorage(val saveFilesDir: String) {
    fun createNote(id: String, content: String) {
        val dir = File(saveFilesDir)
        if (!dir.isDirectory)
            dir.mkdir()

        val targetFile = File(saveFilesDir, "$id.ssnf")
        if (targetFile.exists())
            throw Exception("Note with id $id already exists! Probably, corrupted metadata.")

        targetFile.writeText(content)
    }

    fun updateNote(id: String, content: String) {
        val dir = File(saveFilesDir)
        if (!dir.isDirectory)
            dir.mkdir()

        val targetFile = File(saveFilesDir, "$id.ssnf")
        targetFile.writeText(content)
    }

    fun deleteNote(id: String) {
        val dir = File(saveFilesDir)
        if (!dir.isDirectory)
            return

        val targetFile = File(saveFilesDir, "$id.ssnf")
        if (targetFile.exists())
            targetFile.delete()
    }

    fun getContent(id: String): String {
        val targetFile = File(saveFilesDir, "$id.ssnf")
        return targetFile.readText()
    }
}